### PR TITLE
fix:  deposit should return the deposited amount

### LINKED ISSUE #1104: Purse.deposit() returns new balance
Purse's documentation claims that `deposit(amount)` returns the amount deposited. I think this is correct.

Purse actually returns the new balance of the purse after the deposit.

Edit (Kate, 5/12): Depositing a payment shouldn't implicitly carry the authority to read the balance, so we should not return the new purse balance. Instead, let's go back to returning the payment amount.

### PR DESCRIPTION
**Pull request by dtribble**
closes #1104
Also fixes test to test deposit on a non-empty purse.

### CODE REVIEW

File: packages/ERTP/test/unitTests/test-issuerObj.js [hunk 1]
```diff
@@ -101,28 +101,35 @@ test('issuer.makeEmptyPurse', t => {
     .catch(e => t.assert(false, e));
 });
 
-test('issuer.deposit', t => {
-  t.plan(2);
```
Comment (Chris-Hibbert) [inline packages/ERTP/test/unitTests/test-issuerObj.js:105]: Why drop `t.plan()`? I realize that in the general case it's more work to maintain a number that might change. In this case, the test isn't as likely to require maintenance as the code is. 

I think if the deposit code breaks in a way that causes the the result to be a broken promise, this test will silently succeed, because the `.then()` clauses don't have `onRejected()` clauses. Hopefully, we'd have other tests that would catch that, but this test should also detect failure in that circumstance, I think.

``` js
await E(purse)
    .deposit(payment17, fungible17)
    .then(checkDeposit(fungible0, fungible17));
  await E(purse)
    .deposit(payment25, fungible25)
    .then(checkDeposit(fungible17, fungibleSum));
});	
```

File: packages/ERTP/test/unitTests/test-issuerObj.js [hunk 2]
```diff
@@ -101,28 +101,35 @@ test('issuer.makeEmptyPurse', t => {
...
   const fungible25 = amountMath.make(25);
+  const fungibleSum = amountMath.add(fungible17, fungible25);
 
   const purse = issuer.makeEmptyPurse();
-  const payment = mint.mintPayment(fungible25);
+  const payment17 = mint.mintPayment(fungible17);
+  const payment25 = mint.mintPayment(fungible25);
 
-  const checkDeposit = newPurseBalance => {
+  const checkDeposit = (oldBalance, newBalance) => depositResult => {
```
Comment (katelynsills) [inline packages/ERTP/test/unitTests/test-issuerObj.js]: Could we call this `expectedOldBalance` and `expectedNewBalance`?

**Review by katelynsills — APPROVED** [review]
I think I agree with Chris re: t.plan for now, but otherwise looks good! Just one small comment about naming.

**Comment by katelynsills** [comment]
@dtribble how would you like to move forward with this?

File: packages/ERTP/test/unitTests/test-issuerObj.js (hunk 1 above)
Comment (michaelfig) [reply packages/ERTP/test/unitTests/test-issuerObj.js:105]: The `await`s mean that a missing `.then()` second argument (or `.catch()`) will result in the test returning a rejected promise, and a "test exited without ending" error message.  It needs a `try...catch...finally` around the whole test in order to be less messy when it fails, but I don't see the reason for a `t.plan`.

Comment (dtribble) [reply packages/ERTP/test/unitTests/test-issuerObj.js:105]: Actually, in this test style, the test framework waits for the promise, and if it's breaks, the test fails. So this is actually quite nice. The argument for `plan` is belt and suspenders for making sure that the `asserts` in the abstraction actually get tried. Occasionally people try to do clever tests with async stuff such that some asserts never get tested

Comment (katelynsills) [reply packages/ERTP/test/unitTests/test-issuerObj.js:105]: I think what Michael is saying is that the test failure cascades for every subsequent test, unless you catch it.

**Comment by dtribble** [comment]
Updated.

**Comment by katelynsills** [comment]
zoe-release-0.7.0 was rebased onto master earlier today. I'm still seeing a lot of old changes here though. @dtribble would it be possible to rebase your branch onto zoe-release-0.7.0 again? Or I can do it for you?

**Comment by dtribble** [comment]
Cherry-picked to a clean PR to avoid the rebasing tangle here.
